pyomo/core/plugins/transform/discrete_vars.py

```diff
@@ -44,11 +44,16 @@ def _apply_to(self, model, **kwds):
                 v.setub(bounds[1])
             model.del_component("_relaxed_integer_vars")
             return
+        # True by default, you can specify False if you want
+        descend = kwds.get('descend_into_deactivated_components', 
+                           options.get('descend_into_deactivated_components', 
+                                       True))
```

Comment:
For consistency with the other Pyomo iterators, why not just:
```python
active = kwds.get('active', options.get('active', True))
```

Reply:
I thought about that, but I'm worried that in that case giving the argument `active=False` to the transformation will be really misleading (since from the outside someone maybe doesn't know they're just setting the `active` arg in `component_data_objects`). It is consistent with the other iterators, but I think that the intuitive behavior someone would expect from it is actually what they would get from `active=None`. But... That's kind of true for component_data_objects also. So I'm fine with doing it here too for consistency if you think that's better.

Reply:
Good point.  How about `transform_deactivated_blocks` (after all, only blocks will be descended into)?

Reply:
OK, sounds good. :)